panos/userid.py

```diff
@@ -249,6 +250,9 @@ def register(self, ip, tags):
                 tagelement = ET.SubElement(entry, "tag")
             for tag in tags:
                 member = ET.SubElement(tagelement, "member")
+                if timeout is None:
+                    timeout = "0"
+                member.set("timeout", str(timeout))
```

Comment:
Doing it this way you are unconditionally adding the `timeout` to the tag registration.  A safer way to do this is to only add it if the timeout is actually specified - because I'm not sure if the timeout was always valid or not.

```python
if timeout is not None:
    member.set("timeout", str(timeout))
```

Reply:
Today we tested on  live devices with myself and @bparker-sec 

Test scenarios:
1. Test with a non-zero/valid timeout.
1. Test with timeout set to 0.
1. Test with no timeout value provided.

We tested many permutations of the conditional statement today and the only one we could get to work correctly for all three scenarios is the latest revision.

IIRC the solution you propose worked for scenario 1 and 3, but not for 2 (user specified a timeout of "0"). Please correct me if I am mis-remembering on that one Ben.

If you would like we can repeat the testing with the conditional as you are proposing and add the logs to the PR so you can take a look at the results.

Reply:
I tested again with the changes recommended by @shinmog.  All 3 use cases did work with the recommended changes. 
2021/11/10 17:24:09 vsys1     1.2.3.4                                  timeout_1hour                  2591000  register
XMLAPI                           xml-api
                                                                                                                                                                             
2021/11/10 17:24:09 vsys1     1.1.1.3                                  timeout_set_to_0               0        register
XMLAPI                           xml-api
                                                                                                                                                                             
2021/11/10 17:24:09 vsys1     1.1.1.4                                  timeout_not_set                0        register
XMLAPI                           xml-api

Reply:
Thanks Ben. I've updated the two lines in question and will resolve this issue.